contracts/exchange/api/BaseContract.sol

```diff
@@ -337,4 +349,45 @@ contract BaseContract is AccessControlUpgradeable, ReentrancyGuardUpgradeable {
 
     return pos;
   }
+
+  function _deployDepositProxy(address account) internal returns (address) {
+    bytes32 salt = _getCreate2Salt(account);
+
+    address depositProxy = _deployBeaconProxy(salt);
+    DepositProxy(depositProxy).initialize(address(this), account);
+
+    return depositProxy;
+  }
+
+  function _getCreate2Salt(address accountID) internal pure returns (bytes32 salt) {
+    salt = bytes32(uint256(uint160(accountID)));
+  }
+
+  /// @dev Deploy the beacon proxy for the L2 token, while using ContractDeployer system contract.
+  /// @dev This function uses raw call to ContractDeployer to make sure that exactly `depositProxyProxyBytecodeHash` is used
```

Comment:
i don't understand this part, can you explain?

Reply:
Updated comment